src/doc/src/reference/unstable.md

```diff
@@ -1381,6 +1382,35 @@ The primary use case is to run `cargo rustc --print=cfg` to get config values
 for the appropriate target and influenced by any other RUSTFLAGS.
 
 
+### Different binary name
+
+* Tracking Issue: #1706
+* PR: #9627
+
+This feature allows the user to name their binary without having to obey the 
+restrictions placed on crate names. For example, crate names cannot have numbers
+in them.
```

Comment:
Crate names can have numbers in them, they just can't start with numbers.

Also, I would reword it around using the phrase "the user", since that can be a little ambiguous, and this documentation is addressing "the user", this doesn't need to speak to them in third person.  Perhaps word the start something like "The `different-binary-name` feature allows setting the filename for a binary target without having to obey..."